src/Tracer.ts

```diff
@@ -1,297 +1,79 @@
-/*****
- License
- --------------
- Copyright © 2017 Bill & Melinda Gates Foundation
- The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
- http://www.apache.org/licenses/LICENSE-2.0
- Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
- Contributors
- --------------
- This is the official list of the Mojaloop project contributors for this file.
- Names of the original copyright holders (individuals or organizations)
- should be listed with a '*' in the first column. People who have
- contributed from an organization can be listed under the organization
- that actually holds the copyright for their contributions (see the
- Gates Foundation organization for an example). Those individuals should have
- their names indented and be marked with a '-'. Email address can be added
- optionally within square brackets <email>.
- * Gates Foundation
- - Name Surname <name.surname@gatesfoundation.com>
+import { TraceTags, EventTraceMetadata, EventMessage, TypeSpanContext } from "./model/EventMessage";
 
- - Valentin Genev <valentin.genev@modusbox.com>
+import { getNestedObject } from "./lib/util";
 
- --------------
- ******/
+import { Span, ContextOptions, Recorders } from "./Span"
 
-'use strict'
-
-import { Trace, TraceContext } from "./Trace";
-import { EventLoggingServiceClient, SimpleLoggingServiceClient } from "./transport/EventLoggingServiceClient";
-import { EventType, NullEventAction, AuditEventAction, LogEventAction, TraceEventAction, EventMessage, EventTraceMetadata, IEventTrace, IEventMessage, EventAction, EventStateMetadata, EventMetadata, LogResponseStatus } from "./model/EventMessage";
-
-const Config = require('./lib/config');
-
-/**
- * Logger Options sets the interface for the different logger options, which might be passed to the logger for different actions
- */
-interface LoggerOptions {
-  action?: EventAction,
-  state?: EventStateMetadata
+abstract class ATracer { // TODO consider should we move those to index.js for the API
+  static createSpan: (service: string, tags?: TraceTags, recorders?: Recorders, defaultTagsSetter?: Span['defaultTagsSetter']) => Span
```

Comment:
I would suggest starting static methods with a capital letter. Let me know what you think?